nucliadb/src/nucliadb/search/search/chat/query.py

```diff
@@ -455,32 +454,30 @@ async def run_prequeries(
     x_ndb_client: NucliaDBClientType,
     x_nucliadb_user: str,
     x_forwarded_for: str,
-    generative_model: Optional[str] = None,
-    metrics: RAGMetrics = RAGMetrics(),
+    metrics: Metrics,
 ) -> list[PreQueryResult]:
     """
     Runs simultaneous find requests for each prequery and returns the merged results according to the normalized weights.
     """
     results: list[PreQueryResult] = []
     max_parallel_prequeries = asyncio.Semaphore(settings.prequeries_max_parallel)
 
-    async def _prequery_find(
-        prequery: PreQuery,
-    ):
+    async def _prequery_find(prequery: PreQuery, index: int):
         async with max_parallel_prequeries:
+            prequery_id = prequery.id or str(index)
```

Comment:
Maybe `str(index)` should be `f"prequery-{index}"` or something like that, a bit more clear